interfaces/cython/cantera/yaml2ck.py

```diff
@@ -0,0 +1,714 @@
+    
+    note, comment = thermo_notes(solution_species) # get notes and comments for all species
+
+    thermo_text = []
+    # write data for each species in the Solution object
+    for species in solution_species:
+        composition_string = ''.join([f'{s:2}{int(v):>3}' for s, v in species.composition.items()])
+        
+        name_and_note = f'{species.name} {note[species.name]}'
+        species_string = (comment[species.name] + '\n' +
+            f'{name_and_note:<24}' + # name and date/note field
+            f'{composition_string:<20}' +
+            'G' + # only supports gas phase
+            f'{species.thermo.min_temp:10.3f}' +
+            f'{species.thermo.max_temp:10.3f}' +
+            f'{species.thermo.coeffs[0]:10.3f}' +
+            f'{1:>5}\n') # unused atomic symbols/formula, and blank space
+        
+        # second line has first five coefficients of high-temperature range,
+        # ending with a "2" in column 79
+        species_string += (
+            ''.join([f'{c:15.8e}' for c in species.thermo.coeffs[1:6]]) +
+            f'{2:>5}\n')
+        
+        # third line has the last two coefficients of the high-temperature range,
+        # first three coefficients of low-temperature range, and "3"
+        species_string += (
+            ''.join([f'{c:15.8e}' for c in species.thermo.coeffs[6:8]]) +
+            ''.join([f'{c:15.8e}' for c in species.thermo.coeffs[8:11]]) +
+            f'{3:>5}\n')
+
+        # fourth and last line has the last four coefficients of the
+        # low-temperature range, and "4"
+        species_string += (
+            ''.join([f'{c:15.8e}' for c in species.thermo.coeffs[11:15]]) +
+            f'{4:>20}\n')
+
+        thermo_text.append(species_string)
+    
+    return thermo_text
+
+
+def build_nasa9(solution_species):
+    # Ex: 
+    #   therm NASA9
+    #     200.00  1000.00  6000.00  20000.   3/19/02
+    # e-                Ref-Species. Chase, 1998 3/82.                                
+    #  3 912/98 E   1.00    0.00    0.00    0.00    0.00 0.000548579903          0.000
+    #     298.150   1000.0007 -2.0 -1.0  0.0  1.0  2.0  3.0  4.0  0.0         6197.428
+    #  0.000000000D+00 0.000000000D+00 2.500000000D+00 0.000000000D+00 0.000000000D+00
+    #  0.000000000D+00 0.000000000D+00                -7.453750000D+02-1.172081224D+01
+    #    1000.000   6000.0007 -2.0 -1.0  0.0  1.0  2.0  3.0  4.0  0.0         6197.428
+    #  0.000000000D+00 0.000000000D+00 2.500000000D+00 0.000000000D+00 0.000000000D+00
+    #  0.000000000D+00 0.000000000D+00                -7.453750000D+02-1.172081224D+01
+    #    6000.000  20000.0007 -2.0 -1.0  0.0  1.0  2.0  3.0  4.0  0.0         6197.428
+    #  0.000000000D+00 0.000000000D+00 2.500000000D+00 0.000000000D+00 0.000000000D+00
+    #  0.000000000D+00 0.000000000D+00                -7.453750000D+02-1.172081224D+01
+    #
+    # Could build NASA9 from NASA7 by setting first 2 coefficients to 0
```

Comment:
Optional enhancement:
For NASA9, before going to the `NotImplementedError`, check if it's actually just a NASA7 reformatted to NASA9: if there are only two temperature ranges and the first two coefficients in both are zero, then reformat to NASA-7, else error

Reply:
NASA9 also allow more elements than NASA7. This could cause errors in the future. I'd rather not go down the various rabbit holes of checking the reasons it will or won't work. I would rather point out which species are NASA9 and have the user deal with it however they would like.